Extension/lib/settings/sync-service.js

```diff
@@ -0,0 +1,265 @@
+     * @param localManifest
+     * @returns {{canRead: boolean, canWrite: boolean, sectionsRemoteToLocal: Array, sectionsLocalToRemote: Array}}
+     */
+    var findCompatibility = function (remoteManifest, localManifest) {
+        var canRead = remoteManifest["min-compatible-version"] <= localManifest["min-compatible-version"];
+        var canWrite = remoteManifest["protocol-version"] <= localManifest["protocol-version"];
+
+        var sectionsRemoteToLocal = [];
+        var sectionsLocalToRemote = [];
+
+        if (canRead) {
+            for (var i = 0; i < remoteManifest.sections.length; i++) {
+                var remoteSection = remoteManifest.sections[i];
+                for (var j = 0; j < localManifest.sections.length; j++) {
+                    var localSection = localManifest.sections[j];
+                    if (localSection.name === remoteSection.name) {
+                        var sectionName = localSection.name;
+                        if (localSection.timestamp === remoteSection.timestamp) {
+                            //Do nothing it's synced
+                        } else if (localSection.timestamp < remoteSection.timestamp) {
+                            sectionsRemoteToLocal.push({
+                                name: sectionName,
+                                timestamp: remoteSection.timestamp
+                            });
+                        } else {
+                            if (canWrite) {
+                                sectionsLocalToRemote.push({
+                                    name: sectionName,
+                                    timestamp: localSection.timestamp
+                                });
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return {
+            canRead: canRead,
+            canWrite: canWrite,
+            sectionsRemoteToLocal: sectionsRemoteToLocal,
+            sectionsLocalToRemote: sectionsLocalToRemote
+        }
+    };
+
+    /**
+     * So how does it work:
+     * At first we load manifests for remote and local and then are trying to find can we merge the datasets comparing
+     * protocol versions. Then comparing the timestamps for sections we find the newer data and pick it in some collections
+     * corresponding to the direction: remote-to-local or local-to-remote.
+     *
+     * After that we load selected data sections and process the compatibility result by processSections method,
+     * simply saving the newer sections to remote or local storage.
+     *
+     * As the end we updated manifests with processSections result, setting up the sections timestamps.
+     *
+     * @param callback
+     */
+    var processManifest = function (callback) {
+        var onManifestLoaded = function (remoteManifest) {
+            console.log('Processing remote manifest..');
+
+            if (!remoteManifest) {
+                callback();
```

Comment:
I'm not sure about it.
This file suggested to be something from outside of application.